maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/aether/ClassPathScanningTestCase.java

```diff
@@ -25,38 +25,41 @@
 import org.junit.Test;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.CoreMatchers.nullValue;
+import static org.jboss.shrinkwrap.resolver.impl.maven.aether.ClasspathWorkspaceReader.FLATTENED_POM_PATH_KEY;
+import static org.jboss.shrinkwrap.resolver.impl.maven.aether.ClasspathWorkspaceReader.SUREFIRE_CLASS_PATH_KEY;
 
 /**
+ * Tests {@link ClasspathWorkspaceReader}.
  *
  * @author <a href="kpiwko@redhat.com">Karel Piwko</a>
- *
+ * @author <a href="https://github.com/famod">Falko Modler</a>
+ * @since SHRINKRES-178
  */
-// SHRINKRES-178
 public class ClassPathScanningTestCase {
```

Comment:
Super happy to see this test case being expanded to cover more corner cases. Kudos!

As we are dealing with system properties here - what do you think about using `@Rule` for that? I think it could simplify the test code a bit. In particular, I am thinking of adding https://stefanbirkner.github.io/system-rules/index.html#RestoreSystemProperties

Reply:
That's a good idea! Actually, I have implemented such a rule myself in one of my employer's projects.
I'll have a look at `RestoreSystemProperties`.

Reply:
Done!